Source/ACE/Network/Handlers/CharacterHandler.cs

```diff
@@ -124,33 +124,107 @@ public static void CharacterRestore(ClientMessage message, Session session)
             character.Gender = reader.ReadUInt32();
-            Appearance appearance = Appearance.FromNetowrk(reader);
+            Appearance appearance = Appearance.FromNetwork(reader);
 
             // pull character data from the dat file
             SexCG sex = cg.HeritageGroups[(int)character.Heritage].SexList[(int)character.Gender];
-            
+
             character.MotionTableId = sex.MotionTable;
+            character.SetDataIdProperty(PropertyDataId.MotionTable, character.MotionTableId);
+
             character.SoundTableId = sex.SoundTable;
+            character.SetDataIdProperty(PropertyDataId.SoundTable, character.SoundTableId);
+
             character.PhysicsTableId = sex.PhysicsTable;
+            character.SetDataIdProperty(PropertyDataId.PhysicsEffectTable, character.PhysicsTableId);
+
             character.ModelTableId = sex.SetupID;
+            character.SetDataIdProperty(PropertyDataId.Setup, character.ModelTableId);
+
             character.PaletteId = sex.BasePalette;
+            character.SetDataIdProperty(PropertyDataId.PaletteBase, character.PaletteId);
+
             character.CombatTableId = sex.CombatTable;
-            
-            // not sure how to set these.  Optim says they're in the dat
-            // character.SetDataIdProperty(PropertyDataId.EyesTexture, appearance.Nose);
-            // character.SetDataIdProperty(PropertyDataId.NoseTexture, appearance.Nose);
-            // character.SetDataIdProperty(PropertyDataId.MouthTexture, appearance.Mouth);
+            character.SetDataIdProperty(PropertyDataId.CombatTable, character.CombatTableId);
 
-            // character.SetDataIdProperty(PropertyDataId.HairPalette, appearance.HairColor);
-            // character.SetDataIdProperty(PropertyDataId.EyesPalette, appearance.EyeColor);
-            // character.SetDataIdProperty(PropertyDataId.SkinPalette, appearance.EyeColor);
+            // Check the character scale
+            if (sex.Scale != 100u)
+            {
+                character.SetDoubleProperty(PropertyDouble.DefaultScale, (sex.Scale / 100f)); // Scale is stored as a percentage
+            }
+
+            // Get the hair first, because we need to know if you're bald, and that's the name of that tune!
+            HairStyleCG hairstyle = sex.HairStyleList[Convert.ToInt32(appearance.HairStyle)];
+            bool isBald = hairstyle.Bald;
+
+            // Certain races (Undead, Tumeroks, Others?) have multiple body styles available. This is controlled via the "hair style".
+            if (hairstyle.AlternateSetup > 0)
+            {
+                character.ModelTableId = hairstyle.AlternateSetup;
+                character.SetDataIdProperty(PropertyDataId.Setup, character.ModelTableId);
+            }
 
-            // character.SetDataIdProperty(PropertyDataId.HeadObject, appearance.HairStyle);
+            character.SetDataIdProperty(PropertyDataId.EyesTexture, sex.GetEyeTexture(appearance.Eyes, isBald));
```

Comment:
I'd REALLY like to see a new "EyesTexture" property on AceCharacter that does this on set, as the properties above do.  Same is true for all of these.